Exception thrown when binding argument of supported type.

I'm trying to insert a bean into an sqlite database. The bean has a url field of type URL. When I try inserting it into the database i get this error:

````java
Exception in thread "main" org.jdbi.v3.core.statement.UnableToCreateStatementException: Exception while binding named parameter 'url' [statement:"INSERT INTO book (title, url) VALUES (:title, :url);", rewritten:"INSERT INTO book (title, url) VALUES (:title, :url);", parsed:"ParsedSql{sql='INSERT INTO book (title, url) VALUES (?, ?);', parameters=ParsedParameters{positional=false, parameterNames=[title, url]}}", arguments:{ positional:{}, named:{}, finder:[{lazy bean property arguments "jdbc.Book@5f3a4b84"]}]
	at org.jdbi.v3.core.statement.ArgumentBinder.bindNamed(ArgumentBinder.java:63)
	at org.jdbi.v3.core.statement.ArgumentBinder.bind(ArgumentBinder.java:27)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1443)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:39)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:53)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:41)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$new$2(SqlUpdateHandler.java:56)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$3(SqlUpdateHandler.java:69)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:155)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:29)
	at org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$null$13(SqlObjectFactory.java:162)
	at org.jdbi.v3.core.LazyHandleSupplier.invokeInContext(LazyHandleSupplier.java:73)
	at org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$createInvocationHandler$14(SqlObjectFactory.java:161)
	at com.sun.proxy.$Proxy4.addBook(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jdbi.v3.core.OnDemandExtensions.lambda$null$0(OnDemandExtensions.java:60)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:464)
	at org.jdbi.v3.core.OnDemandExtensions.lambda$create$1(OnDemandExtensions.java:57)
	at com.sun.proxy.$Proxy4.addBook(Unknown Source)
````
Here are my dao interface and bean class

````java
public interface BookDao {
    @SqlUpdate("CREATE TABLE IF NOT EXISTS book (" +
            "id INTEGER PRIMARY KEY, " +
            "title TEXT, " +
            "url TEXT);")
    void createTableIfNotExits();

    @SqlQuery("SELECT * FROM book;")
    @RegisterConstructorMapper(Book.class)
    List<Book> getAllBooks();

    @SqlUpdate("INSERT INTO book (title, url) VALUES (:title, :url);")
    boolean addBook(@BindBean Book book);

    @SqlUpdate("DELETE FROM book WHERE id=:id;")
    boolean removeBooks(@BindBean Book book);
}
````

````java
public class Book {
    private final int id;
    private String title;
    private URL url;

    public Book(@ColumnName("id") int id, @ColumnName("title") String title, @ColumnName("url") URL url) {
        this.id = id;
        this.title = title;
        this.url = url;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public URL getUrl() {
        return url;
    }
}
````

According to the documentation, URL is a supported type so i'm thinking it would automatically be mapped to a string for me. Am I doing something wrong?

````java
Caused by: java.sql.SQLException: not implemented by SQLite JDBC driver
	at org.sqlite.jdbc3.JDBC3PreparedStatement.unused(JDBC3PreparedStatement.java:499)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.setURL(JDBC3PreparedStatement.java:516)
	at org.jdbi.v3.core.argument.BuiltInArgumentFactory$BuiltInArgument.apply(BuiltInArgumentFactory.java:192)
	at org.jdbi.v3.core.statement.ArgumentBinder.bindNamed(ArgumentBinder.java:60)
	... 22 more
````

--- a/docs/pom.xml
+++ b/docs/pom.xml
@@ -108,6 +108,11 @@
       </dependency>
 
       <dependency>
+          <groupId>org.jdbi</groupId>
+          <artifactId>jdbi3-sqlite</artifactId>
+      </dependency>
+
+      <dependency>
           <groupId>com.google.code.findbugs</groupId>
           <artifactId>findbugs-annotations</artifactId>
       </dependency>
@@ -143,7 +148,7 @@
          <artifactId>kotlin-test</artifactId>
          <scope>test</scope>
       </dependency>
-      
+
       <dependency>
           <groupId>org.jetbrains.kotlin</groupId>
           <artifactId>kotlin-test-junit</artifactId>

--- a/docs/src/adoc/index.adoc
+++ b/docs/src/adoc/index.adoc
@@ -3947,6 +3947,33 @@ Bear in mind:
   `TupleMappers` must be configured properly i.e. all non row-mapped
   tuple indices must be provided with a column configuration!
 
+=== SQLite
+The *jdbi3-sqlite* plugin provides support for using the
+link:https://bitbucket.org/xerial/sqlite-jdbc/[SQLite JDBC Driver^]
+with Jdbi.
+
+The plugin configures mapping for the Java *URL* type which is not
+supported by driver.
+
+To use this plugin, first add a Maven dependency:
+
+[source,xml,subs="specialchars"]
+----
+<dependency>
+  <groupId>org.jdbi</groupId>
+  <artifactId>jdbi3-sqlite</artifactId>
+</dependency>
+----
+
+Then install the plugin into your `Jdbi` instance.
+
+[source,java,indent=0]
+----
+Jdbi jdbi = Jdbi.create("jdbc:sqlite:database")
+                .installPlugin(new SQLitePlugin());
+----
+
+
 
 == Advanced Topics
 

--- a/pom.xml
+++ b/pom.xml
@@ -468,5 +468,6 @@
         <module>bom</module>
         <module>noparameters</module>
         <module>vavr</module>
+        <module>sqlite</module>
     </modules>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -190,6 +190,11 @@
             </dependency>
             <dependency>
                 <groupId>org.jdbi</groupId>
+                <artifactId>jdbi3-sqlite</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.jdbi</groupId>
                 <artifactId>jdbi3-jodatime2</artifactId>
                 <version>${project.version}</version>
             </dependency>

--- a/sqlite/pom.xml
+++ b/sqlite/pom.xml
@@ -1,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>jdbi3-parent</artifactId>
+        <groupId>org.jdbi</groupId>
+        <version>3.0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jdbi3-sqlite</artifactId>
+    <name>jdbi3 sqlite</name>
+    <description>
+        Jdbi sqlite support
+    </description>
+    <url>http://jdbi.org/</url>
+
+    <licenses>
+        <license>
+            <name>Apache License 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.html</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.jdbi</groupId>
+            <artifactId>jdbi3-core</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.xerial</groupId>
+            <artifactId>sqlite-jdbc</artifactId>
+            <version>3.21.0.1</version>
+        </dependency>
+    </dependencies>
+</project>

--- a/sqlite/pom.xml
+++ b/sqlite/pom.xml
@@ -47,6 +47,7 @@
             <groupId>org.xerial</groupId>
             <artifactId>sqlite-jdbc</artifactId>
             <version>3.21.0.1</version>
+            <scope>test</scope>
         </dependency>
     </dependencies>
 </project>

--- a/sqlite/pom.xml
+++ b/sqlite/pom.xml
@@ -1,4 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+~   Licensed under the Apache License, Version 2.0 (the "License");
+~   you may not use this file except in compliance with the License.
+~   You may obtain a copy of the License at
+~
+~   http://www.apache.org/licenses/LICENSE-2.0
+~
+~   Unless required by applicable law or agreed to in writing, software
+~   distributed under the License is distributed on an "AS IS" BASIS,
+~   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~   See the License for the specific language governing permissions and
+~   limitations under the License.
+-->
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/SQLitePlugin.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/SQLitePlugin.java
@@ -19,7 +19,7 @@ import org.jdbi.v3.core.spi.JdbiPlugin;
 import java.net.URL;
 
 /**
- * Jdbc plugin for SQLite
+ * Jdbi plugin for SQLite.
  */
 public class SQLitePlugin implements JdbiPlugin {
     @Override

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/SQLitePlugin.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/SQLitePlugin.java
@@ -1,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlite3;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.spi.JdbiPlugin;
+
+/**
+ * Jdbc plugin for SQLite
+ */
+public class SQLitePlugin implements JdbiPlugin {
+    @Override
+    public void customizeJdbi(Jdbi jdbi) {
+        jdbi.registerArgument(new URLArgumentFactory());
+    }
+}

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLArgumentFactory.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLArgumentFactory.java
@@ -1,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jdbi.v3.sqlite3;
+
+import org.jdbi.v3.core.argument.AbstractArgumentFactory;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+import java.net.URL;
+import java.sql.Types;
+
+/**
+ * Build URL objects as strings.
+ */
+public class URLArgumentFactory extends AbstractArgumentFactory<URL> {
+    /**
+     * Constructs an {@link ArgumentFactory} for type {@code URL}.
+     */
+    public URLArgumentFactory() {
+        super(Types.VARCHAR);
+    }
+
+    @Override
+    protected Argument build(URL url, ConfigRegistry config) {
+        return (position, statement, ctx) -> statement.setString(position, url.toString());
+    }
+}

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLArgumentFactory.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLArgumentFactory.java
@@ -25,7 +25,7 @@ import java.sql.Types;
 /**
  * Build URL objects as strings.
  */
-public class URLArgumentFactory extends AbstractArgumentFactory<URL> {
+class URLArgumentFactory extends AbstractArgumentFactory<URL> {
     /**
      * Constructs an {@link ArgumentFactory} for type {@code URL}.
      */

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
@@ -22,7 +22,7 @@ import java.net.URL;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 
-public class URLColumnMapper implements ColumnMapper<URL> {
+class URLColumnMapper implements ColumnMapper<URL> {
 
     @Override
     public URL map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
@@ -13,18 +13,26 @@
  */
 package org.jdbi.v3.sqlite3;
 
-import org.jdbi.v3.core.Jdbi;
-import org.jdbi.v3.core.spi.JdbiPlugin;
+import org.jdbi.v3.core.mapper.ColumnMapper;
+import org.jdbi.v3.core.statement.StatementContext;
 
+import java.net.MalformedURLException;
+import java.net.URI;
 import java.net.URL;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 
-/**
- * Jdbc plugin for SQLite
- */
-public class SQLitePlugin implements JdbiPlugin {
+public class URLColumnMapper implements ColumnMapper<URL> {
+
     @Override
-    public void customizeJdbi(Jdbi jdbi) {
-        jdbi.registerArgument(new URLArgumentFactory());
-        jdbi.registerColumnMapper(new URLColumnMapper());
+    public URL map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
+        URL url = null;
+
+        try {
+            url = URI.create(r.getString(columnNumber)).toURL();
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+        return url;
     }
 }

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
@@ -29,7 +29,7 @@ class URLColumnMapper implements ColumnMapper<URL> {
         URL url = null;
 
         try {
-            url = URI.create(r.getString(columnNumber)).toURL();
+            url = new URL(r.getString(columnNumber));
         } catch (MalformedURLException e) {
             throw new SQLException(e);
         }

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
@@ -31,7 +31,7 @@ public class URLColumnMapper implements ColumnMapper<URL> {
         try {
             url = URI.create(r.getString(columnNumber)).toURL();
         } catch (MalformedURLException e) {
-            e.printStackTrace();
+            throw new SQLException(e);
         }
         return url;
     }

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/URLColumnMapper.java
@@ -26,13 +26,10 @@ class URLColumnMapper implements ColumnMapper<URL> {
 
     @Override
     public URL map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
-        URL url = null;
-
         try {
-            url = new URL(r.getString(columnNumber));
+            return new URL(r.getString(columnNumber));
         } catch (MalformedURLException e) {
             throw new SQLException(e);
         }
-        return url;
     }
 }

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/package.html
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/package.html
@@ -1,0 +1,22 @@
+<!--
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<html>
+<body>
+<p>
+    The <code>sqlite</code> plugin provides improved support for
+    <code>java.net.URL</code>.
+</p>
+</body>
+</html>

--- a/sqlite/src/main/java/org/jdbi/v3/sqlite3/package.html
+++ b/sqlite/src/main/java/org/jdbi/v3/sqlite3/package.html
@@ -1,4 +1,5 @@
 <!--
+
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
     You may obtain a copy of the License at

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -48,8 +48,7 @@ public class TestUrls {
     @Test
     public void testInsertUrlSuccessful() throws MalformedURLException {
         String goolgeString = "http://www.google.com";
-        URL googleUrl = null;
-        googleUrl = URI.create(goolgeString).toURL();
+        URL googleUrl = new URL(goolgeString);
 
         handle.createUpdate("INSERT INTO foo VALUES (:url)")
                 .bind("url", googleUrl)

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -46,8 +46,8 @@ public class TestUrls {
 
     @Test
     public void testInsertUrlSuccessful() throws MalformedURLException {
-        String goolgeString = "http://www.google.com";
-        URL googleUrl = new URL(goolgeString);
+        String googleString = "http://www.google.com";
+        URL googleUrl = new URL(googleString);
 
         handle.createUpdate("INSERT INTO foo VALUES (:url)")
                 .bind("url", googleUrl)
@@ -67,7 +67,7 @@ public class TestUrls {
     }
 
     @Test
-    public void testInserUrlUsingBindByType() throws MalformedURLException {
+    public void testInsertUrlUsingBindByType() throws MalformedURLException {
         URL githubUrl = new URL("http://www.github.com");
 
         handle.createUpdate("INSERT INTO foo VALUES (:url)")

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -78,4 +78,13 @@ public class TestUrls {
         Assertions.assertThat(dbUrl).hasToString(githubUrl.toString());
     }
 
+    @Test(expected = ResultSetException.class)
+    public void testInsertNullUrlUsingBindByType() {
+        handle.createUpdate("INSERT INTO foo VALUES (:url)")
+                .bindByType("url", null, URL.class)
+                .execute();
+
+        handle.createQuery("SELECT url FROM foo").mapTo(URL.class).findOnly();
+    }
+
 }

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -36,7 +36,6 @@ public class TestUrls {
         jdbi.installPlugin(new SQLitePlugin());
         handle = jdbi.open();
         handle.useTransaction(handle -> {
-            handle.execute("DROP TABLE IF EXISTS foo");
             handle.execute("CREATE TABLE foo(url URL);");
         });
     }
@@ -47,14 +46,10 @@ public class TestUrls {
     }
 
     @Test
-    public void testInsertUrlSuccessful() {
+    public void testInsertUrlSuccessful() throws MalformedURLException {
         String goolgeString = "http://www.google.com";
         URL googleUrl = null;
-        try {
-            googleUrl = URI.create(goolgeString).toURL();
-        } catch (MalformedURLException e) {
-            e.printStackTrace();
-        }
+        googleUrl = URI.create(goolgeString).toURL();
 
         handle.createUpdate("INSERT INTO foo VALUES (:url)")
                 .bind("url", googleUrl)

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -37,7 +37,7 @@ public class TestUrls {
         handle = jdbi.open();
         handle.useTransaction(handle -> {
             handle.execute("DROP TABLE IF EXISTS foo");
-            handle.execute("CREATE TABLE foo(url TEXT);");
+            handle.execute("CREATE TABLE foo(url URL);");
         });
     }
 
@@ -60,7 +60,7 @@ public class TestUrls {
                 .bind("url", googleUrl)
                 .execute();
 
-        Assertions.assertThat(handle.createQuery("SELECT * FROM foo").mapTo(String.class).findOnly()).isEqualTo(goolgeString);
+        Assertions.assertThat(handle.createQuery("SELECT * FROM foo").mapTo(URL.class).findOnly()).isEqualTo(googleUrl);
     }
 
 }

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -1,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlite3;
+
+import org.assertj.core.api.Assertions;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.HandleConsumer;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlite3.SQLitePlugin;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+public class TestUrls {
+
+    Handle handle;
+
+    @Before
+    public void setUp() throws Exception {
+        Jdbi jdbi = Jdbi.create("jdbc:sqlite::memory:");
+        jdbi.installPlugin(new SQLitePlugin());
+        handle = jdbi.open();
+        handle.useTransaction(handle -> {
+            handle.execute("DROP TABLE IF EXISTS foo");
+            handle.execute("CREATE TABLE foo(url TEXT);");
+        });
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        handle.close();
+    }
+
+    @Test
+    public void testInsertUrlSuccessful() {
+        String goolgeString = "http://www.google.com";
+        URL googleUrl = null;
+        try {
+            googleUrl = URI.create(goolgeString).toURL();
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+
+        handle.createUpdate("INSERT INTO foo VALUES (:url)")
+                .bind("url", googleUrl)
+                .execute();
+
+        Assertions.assertThat(handle.createQuery("SELECT * FROM foo").mapTo(String.class).findOnly()).isEqualTo(goolgeString);
+    }
+
+}

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -17,6 +17,7 @@ import org.assertj.core.api.Assertions;
 import org.jdbi.v3.core.Handle;
 import org.jdbi.v3.core.HandleConsumer;
 import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.result.ResultSetException;
 import org.jdbi.v3.sqlite3.SQLitePlugin;
 import org.junit.After;
 import org.junit.Before;
@@ -52,8 +53,29 @@ public class TestUrls {
                 .bind("url", googleUrl)
                 .execute();
 
-        URL actualUrl = handle.createQuery("SELECT * FROM foo").mapTo(URL.class).findOnly();
+        URL actualUrl = handle.createQuery("SELECT url FROM foo").mapTo(URL.class).findOnly();
         Assertions.assertThat(actualUrl).hasToString(googleUrl.toString());
+    }
+
+    @Test(expected = ResultSetException.class)
+    public void testMapNullUrlThrowsException() {
+        handle.createUpdate("INSERT INTO foo VALUES (:url)")
+                .bind("url", ((URL) null))
+                .execute();
+
+        handle.createQuery("SELECT url FROM foo").mapTo(URL.class).findOnly();
+    }
+
+    @Test
+    public void testInserUrlUsingBindByType() throws MalformedURLException {
+        URL githubUrl = new URL("http://www.github.com");
+
+        handle.createUpdate("INSERT INTO foo VALUES (:url)")
+                .bindByType("url", githubUrl, URL.class)
+                .execute();
+
+        URL dbUrl = handle.createQuery("SELECT * FROM foo").mapTo(URL.class).findOnly();
+        Assertions.assertThat(dbUrl).hasToString(githubUrl.toString());
     }
 
 }

--- a/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
+++ b/sqlite/src/test/java/org/jdbi/v3/sqlite3/TestUrls.java
@@ -35,9 +35,7 @@ public class TestUrls {
         Jdbi jdbi = Jdbi.create("jdbc:sqlite::memory:");
         jdbi.installPlugin(new SQLitePlugin());
         handle = jdbi.open();
-        handle.useTransaction(handle -> {
-            handle.execute("CREATE TABLE foo(url URL);");
-        });
+        handle.useTransaction(handle -> handle.execute("CREATE TABLE foo(url URL);"));
     }
 
     @After
@@ -54,7 +52,8 @@ public class TestUrls {
                 .bind("url", googleUrl)
                 .execute();
 
-        Assertions.assertThat(handle.createQuery("SELECT * FROM foo").mapTo(URL.class).findOnly()).isEqualTo(googleUrl);
+        URL actualUrl = handle.createQuery("SELECT * FROM foo").mapTo(URL.class).findOnly();
+        Assertions.assertThat(actualUrl).hasToString(googleUrl.toString());
     }
 
 }

